### PR TITLE
Display transaction error reason

### PR DESCRIPTION
**Pull request by msavir**
Displays the transaction failed reason on the snack bar

image

### CODE REVIEW

File: src/pages/Governance/components/snackbar/FailureSnackbar.tsx [hunk 1]
```diff
@@ -26,16 +29,19 @@ export default function FailureSnackbar({
...
             component={RRD.Link}
             to={`/txn/${transactionHash}`}
             color="inherit"
             target="_blank"
           >
             {transactionHash}
-          </Link>
-          {"."} Please try again.
+          </Link>{" "}
+          Failed{" "}
```
Comment (zihan-aptos) [inline src/pages/Governance/components/snackbar/FailureSnackbar.tsx:41]:
```suggestion
          failed{" "}
```

File: src/pages/Governance/components/snackbar/FailureSnackbar.tsx [hunk 2]
```diff
@@ -2,15 +2,18 @@ import * as RRD from "react-router-dom";
 import React from "react";
 import {Snackbar, Alert, Typography, Link} from "@mui/material";
 import {CloseAction} from "./TransactionResponseSnackbar";
+import {Types} from "aptos";
 
 type FailureSnackbarProps = {
   transactionHash: string;
   onCloseSnackbar: () => void;
+  data: Types.Transaction | undefined;
```
Comment (zihan-aptos) [inline src/pages/Governance/components/snackbar/FailureSnackbar.tsx:10]: Is `transactionHash` in `data`? If so, can we just pass in `data` as prop?

**Review by zihan-aptos — APPROVED** [review]

File: src/pages/Governance/components/snackbar/FailureSnackbar.tsx [hunk 3]
```diff
@@ -26,16 +29,19 @@ export default function FailureSnackbar({
...
             color="inherit"
             target="_blank"
           >
             {transactionHash}
-          </Link>
-          {"."} Please try again.
+          </Link>{" "}
+          Failed{" "}
+          {data && "vm_status" in data && data.vm_status
+            ? `with ${data.vm_status}`
```
Comment (zihan-aptos) [inline src/pages/Governance/components/snackbar/FailureSnackbar.tsx:43]: Wondering if it's better to add "..." outside the vm_status to make it clear that this part is the message.

Comment (msavir) [reply src/pages/Governance/components/snackbar/FailureSnackbar.tsx:43]: not completely sure what you mean here..?

Comment (zihan-aptos) [reply src/pages/Governance/components/snackbar/FailureSnackbar.tsx:43]: Sorry, I meant something like this 
```suggestion
            ? `with "${data.vm_status}"`
```

**Pull request merged**